common/src/main/java/io/novafoundation/nova/common/utils/recyclerView/expandable/ExpandableItemAnimator.kt

```diff
@@ -0,0 +1,292 @@
+            return true
+        }
+
+        val item = (holder as ExpandableChildViewHolder).expandableItem!!
+
+        // Reset move state helps clear translationY when animation is being to be canceled
+        if (pendingMoveAnimations.contains(holder)) {
+            holder.itemView.animate().cancel()
+
+            resetMoveState(holder)
+        }
+
+        if (pendingRemoveAnimations.contains(holder)) {
+            holder.itemView.animate().cancel()
+        } else {
+            preAddImpl(holder)
+        }
+
+        if (item.groupId !in addAnimations) addAnimations[item.groupId] = mutableListOf()
+        addAnimations[item.groupId]?.add(holder)
+
+        expandableAnimator.prepareAnimationToState(item.groupId, ExpandableAnimationItemState.Type.EXPANDING)
+
+        return true
+    }
+
+    override fun animateRemove(holder: ViewHolder): Boolean {
+        val notPreparedForAnimation = !preparedForAnimation
+        val notExpandableChildItem = holder !is ExpandableChildViewHolder || holder.expandableItem == null
+        if (notPreparedForAnimation || notExpandableChildItem) {
+            dispatchRemoveFinished(holder)
+            return true
+        }
+
+        val item = (holder as ExpandableChildViewHolder).expandableItem!!
+
+        // Reset move state helps clear translationY when animation is being to be canceled
+        if (pendingMoveAnimations.contains(holder)) {
+            holder.itemView.animate().cancel()
+
+            resetMoveState(holder)
+        }
+
+        if (pendingAddAnimations.contains(holder)) {
+            holder.itemView.animate().cancel()
+        } else {
+            preRemoveImpl(holder)
+        }
+
+        if (item.groupId !in removeAnimations) removeAnimations[item.groupId] = mutableListOf()
+        removeAnimations[item.groupId]?.add(holder)
+
+        expandableAnimator.prepareAnimationToState(item.groupId, ExpandableAnimationItemState.Type.COLLAPSING)
+
+        return true
+    }
+
+    override fun animateMove(holder: ViewHolder, fromX: Int, fromY: Int, toX: Int, toY: Int): Boolean {
+        val notPreparedForAnimation = !preparedForAnimation
+        if (notPreparedForAnimation || holder !is ExpandableBaseViewHolder<*>) {
+            dispatchMoveFinished(holder)
+            return true
+        }
+
+        // Reset add state helps clear alpha and scale when animation is being to be canceled
+        if (pendingAddAnimations.contains(holder)) {
+            holder.itemView.animate().cancel()
+            resetAddState(holder)
+        }
+
+        // Reset remove state helps clear alpha and scale when animation is being to be canceled
+        if (pendingRemoveAnimations.contains(holder)) {
+            holder.itemView.animate().cancel()
+            resetRemoveState(holder)
+        }
+
+        if (pendingMoveAnimations.contains(holder)) {
+            holder.itemView.animate().cancel()
+        }
+
+        preMoveImpl(holder, fromY, toY)
```

Comment:
Is there a specific reason `preMoveImpl` is one regardless of `if` checks above but for `add` and `remove` we only do `preRemoveImpl` and ` preAddImpl` in `else` branch?